emmet-core/emmet/core/qchem/calculation.py

```diff
@@ -0,0 +1,437 @@
+# as QChem data objects
+
+
+class CalculationInput(BaseModel):
+    """
+    Document defining QChem calculation inputs.
+    """
+
+    initial_molecule: Dict[str, Any] = Field(
+        None, description="input molecule geometry before the QChem calculation"
+    )
+
+    # parameters: Dict[str, Any] = Field(
+    #     None, description = "Parameters from a previous QChem calculation."
+    # )
+
+    charge: int = Field(None, description="The charge of the input molecule")
+
+    rem: Dict[str, Any] = Field(
+        None,
+        description="The rem dict of the input file which has all the input parameters",
+    )
+
+    job_type: str = Field(
+        None, description="The type of QChem calculation being performed"
+    )
+
+    opt: Dict[str, Any] = Field(
+        None,
+        description="A dictionary of opt section. For instance atom constraints and fixed atoms. Go to QCInput definition for more details.",
+    )
+
+    pcm: Dict[str, Any] = Field(
+        None, description="A dictionary for the PCM solvent details if used"
+    )
+
+    solvent: Dict[str, Any] = Field(
+        None,
+        description="The solvent parameters used if the PCM solvent model has been employed",
+    )
+
+    smx: Dict[str, Any] = Field(
+        None,
+        description="A dictionary for the solvent parameters if the SMD solvent method has been employed",
+    )
+
+    vdw_mode: str = Field(
+        None,
+        description="Either atomic or sequential. Used when custon van der Waals radii are used to construct pcm cavity",
+    )
+
+    van_der_waals: Dict[str, Any] = Field(
+        None, description="The dictionary of the custom van der Waals radii if used"
+    )
+
+    scan_variables: Dict[str, Any] = Field(
+        None,
+        description="The dictionary of scan variables for torsions or bond stretches",
+    )
+
+    tags: Union[Dict[str, Any], str] = Field(
+        None, description="Any tags associated with the QChem calculation."
+    )
+
+    @classmethod
+    def from_qcinput(cls, qcinput: QCInput) -> "CalculationInput":
+        """
+        Create a QChem input document from a QCInout object.
+
+        Parameters
+        ----------
+        qcinput
+            A QCInput object.
+
+        Returns
+        --------
+        CalculationInput
+            The input document.
+        """
+
+        return cls(
+            initial_molecule=qcinput.molecule.as_dict(),
+            charge=int(qcinput.molecule.as_dict()["charge"])
+            if qcinput.molecule.as_dict()
+            else None,
+            rem=qcinput.rem,
+            job_type=qcinput.rem.get("job_type", None),
+            opt=qcinput.opt,
+            pcm=qcinput.pcm,
+            solvent=qcinput.solvent,
+            smx=qcinput.smx,
+            vdw_mode=qcinput.vdw_mode,
+            scan_variables=qcinput.scan,
+            van_der_waals=qcinput.van_der_waals,
+        )
+
+
+class CalculationOutput(BaseModel):
```

Comment:
Again, we should probably have a conversation about what details should be on the top level. Also, is this meant to represent the entire QCOutput, or just the main "output" data that's in a task document?